### DISCUSSION TITLE
Pin Assignment for any ESP32 and LAN8720

### DISCUSSION THREAD

**Discussion by martinmmi** [espressif/arduino-esp32#7739] [Q&A]
Hello, i am Martin from Germany and i have one basic question to this part of the repository:

https://github.com/espressif/arduino-esp32/blob/f1b06d2a74acdef190ac16e83d48ab619f557e73/libraries/Ethernet/examples/ETH_LAN8720/ETH_LAN8720.ino

Since few weeks I try to connect an **LAN8720** to an **LILYOGO ESP32**. Its not so easy because the LAN8720 Board needs much external signals from the esp.

**For Example:**

- MDC / MDIO (I²C Bus)
- CRS
- RX1
- RX0
- nINT (CLOCK)
- etc.

**Now my question:** Where can i do the pin assignments in the [WebServer_WT32_ETH01] Lib? Because every pinout is different on microcontrollers. Maybe for the example: AdvancedWebServer.ino

I have that controller concrete:
http://www.lilygo.cn/prod_view.aspx?TypeId=50003&Id=1271&FId=t3:50003:3

Thank you so much for reply in advance. I'll show my appreciation if this works.

**Comment by AmauriSarto** [comment]
Hello,


#ifndef ETH_PHY_ADDR
#define ETH_PHY_ADDR 0
#endif
#ifndef ETH_PHY_TYPE
#define ETH_PHY_TYPE ETH_PHY_LAN8720
#endif
#ifndef ETH_PHY_POWER
#define ETH_PHY_POWER -1
#endif
#ifndef ETH_PHY_MDC
#define ETH_PHY_MDC 23
#endif
#ifndef ETH_PHY_MDIO
#define ETH_PHY_MDIO 18
#endif


https://github.com/espressif/arduino-esp32/blob/f1b06d2a74acdef190ac16e83d48ab619f557e73/libraries/Ethernet/src/ETH.h

Em sex., 20 de jan. de 2023 às 12:22, martinmmi ***@***.***>
escreveu:
